semmc-aarch32/src/SemMC/Architecture/ARM/ASL.hs

```diff
@@ -306,7 +323,39 @@ symFnToParamFormula sym (UFBundle { getGPR, getSIMD, initGPRs, initSIMDs, initMe
         expr'' <- WI.applySymFn sym updateMemory (Ctx.singleton expr') 
         return $ GlobalParameter (SF.LiteralParameter (ARM.Location ASL.MemoryRef)) expr'')
 
-  defs <- MapF.fromList . catMaybes . FC.toListFC (\(Const c) -> c) <$> Ctx.zipWithM filterTrivial gbvars glbParams
+  defs' <- MapF.fromList . catMaybes . FC.toListFC (\(Const c) -> c) <$> Ctx.zipWithM filterTrivial gbvars glbParams
+
+  gprsExpr' <- do
+    gprsExpr <- return $ ASL.sGPRs structExprs
+    case gprsExpr of
+      WB.NonceAppExpr nae
+        | WB.FnApp symFn' _args <- WB.nonceExprApp nae
+        , Just Refl <- testEqualitySymFn symFn' initGPRs
+        -> return Nothing
+      _ -> Just <$> WI.applySymFn sym updateGPRs (Ctx.singleton gprsExpr)
+
+  simdsExpr' <- do
+    simdsExpr <- return $ ASL.sSIMDs structExprs
+    case simdsExpr of
+      WB.NonceAppExpr nae
+        | WB.FnApp symFn' _args <- WB.nonceExprApp nae
+        , Just Refl <- testEqualitySymFn symFn' initSIMDs
+        -> return Nothing
+      _ -> Just <$> WI.applySymFn sym updateSIMDs (Ctx.singleton simdsExpr)
+
+  let statefulOp = case (gprsExpr', simdsExpr') of
+        (Nothing, Nothing) -> False
+        _ -> True
+
+  noopApplied <- WI.applySymFn sym noop Ctx.empty
+  let gprsExpr = fromMaybe noopApplied gprsExpr'
+  let simdsExpr = fromMaybe noopApplied simdsExpr'
+
+  dummyExpr <- WI.applySymFn sym joinUnits (Ctx.empty Ctx.:> gprsExpr Ctx.:> simdsExpr)
+
+  let defs = case statefulOp of
+        True ->  MapF.insert (SF.LiteralParameter (ARM.Location (ASL.knownGlobalRef @"__DummyValue"))) dummyExpr defs'
```

Comment:
Can we get a note about where dummy values arise in the translation? I assume it is something we add?